controllers/user/lib/account-schemas.js

```diff
@@ -0,0 +1,161 @@
+'use strict';
+const fs = require('fs');
+const path = require('path');
+const get = require('lodash/fp/get');
+const Joi = require('@hapi/joi');
+
+/**
+ * List of the most commonly used passwords. Specifically to 10k, filtered by min-length
+ * awk 'length($0)>9' full-list.txt > common-passwords.txt
+ * @see https://github.com/danielmiessler/SecLists/tree/master/Passwords/Common-Credentials
+ */
+const commonPasswords = fs
+    .readFileSync(path.resolve(__dirname, './common-passwords.txt'), 'utf8')
+    .toString()
+    .split('\n');
+
+const MIN_PASSWORD_LENGTH = 10;
+
+const username = Joi.string()
+    .email()
+    .required();
+
+const passwordSchema = Joi.string()
+    .min(MIN_PASSWORD_LENGTH) // Min characters
+    .invalid(Joi.ref('username')) // Must not equal username
+    .invalid(commonPasswords) // Must not be in common passwords list
+    .required();
+
+const passwordConfirmationSchema = Joi.string()
+    .valid(Joi.ref('password')) // Must match password
+    .required();
+
+const MESSAGES = {
+    emailInvalid(locale) {
+        return get(locale)({
+            en: 'Enter a valid email address',
+            cy: ''
```

Comment:
The `/user/register` endpoint is public right now (!) so perhaps we should get these labels translated now, independent of the AFA launch (or we might forget)?

Reply:
... Thinking about it, the whole thing is English-only so I guess we should raise a ticket for the next sprint to get this translated (perhaps once Lily and Scott have finalised the copy for the account emails?)